src/tensor_parallel/slicer_wrapper.py

```diff
@@ -217,12 +220,28 @@ def apply_action(input: torch.Tensor, action: Action, *, rank: int, world_size:
         return torch.tensor_split(input, world_size, dim=dim)[rank]
     if action_type == "scale":
         return input / world_size
-    if action_type == "scale_int":
-        assert input % world_size == 0
-        return input // world_size
     raise Exception(f"unexpected action {action_type}; supported actions: split, scale, or custom user-defined")
 
 
+def apply_inverse_action(tensors: Sequence[torch.Tensor], action: StateAction, world_size: int) -> torch.Tensor:
+    assert isinstance(action, str) or isinstance(
+        action, tuple
+    ), "No inverse state action specified for custom action. Can't aggregate shards"
```

Comment:
- Maybe better "Can't undo tensor_parallel"? (why: user who's getting this assertion from inside the module may not know what this is)
- Maybe better to throw RuntimeError or NotImplementedError?
  - Why: AssertionError is associated with "the library forked up", and the above errors are more general
  - THis is a nitpick. If you consciously prefer assertion error, keep it :)
- Can we still save __sharded__ model for use in a sharded way later?
    - not super important, but i think some workloads can use that. I personally did. If you wanna ignore it, i won't object